src/components/Navigator.js

```diff
@@ -197,7 +223,7 @@ class Navigator extends React.Component {
       alert('failed to create new file');
       return;
     }
-    const updatedSourceId = sourceFileType ? newFile.id : this.props.currentOpenFileId.sourceId;
+    const updatedSourceId = sourceFileType ? newFile.id : this.state.selectedFileId;
```

Comment:
what's the reason for this change?

Reply:
idk how this worked earlier. But basically above this code we close the "current file" when opening a new one. Then currentOpenFileId is set to None. But i already have it stored in the state for highlighting purposes, so i just used that.

Reply:
The way it worked earlier was when it closed the open file, the state change was propagated via redux, and then in the componentDidUpdate, it updates the value of the displayed names. I think this state change flow is much more consistent - see my previous comment about not having duplicate sources of state.